streaming-connect-sink/src/main/java/com/adobe/platform/streaming/sink/impl/AEPPublisher.java

```diff
@@ -87,7 +94,27 @@ public void publishData(List<Pair<String, SinkRecord>> messages) throws AEPStrea
       );
 
       count++;
-      LOG.debug("Successfully published data to Adobe Experience Platform: {}", response);
+
+      if (!response.isEmpty() && response.hasNonNull(RESPONSE_CODE_KEY) &&
+        response.get(RESPONSE_CODE_KEY).asInt() == HttpStatus.SC_MULTI_STATUS) {
+        JsonNode publishMessagesResponses = response.get(RESPONSES_KEY);
+        for (JsonNode messageResponse : publishMessagesResponses) {
+          if (messageResponse.hasNonNull(STATUS_KEY)) {
```

Comment:
Should we also check if status code is 4xx or if status code is present that means its a failure from inlet ?

Reply:
Yeah if status code is present that means there is an error in processing that message. In case of success no status is present in the responses.